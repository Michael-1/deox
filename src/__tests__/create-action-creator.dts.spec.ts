import { createActionCreator } from '../create-action-creator'

// @dts-jest:group createActionCreator

const a = createActionCreator('A')

// @dts-jest:pass:snap
createActionCreator('A')

// @dts-jest:fail
a(0)

// @dts-jest:pass:snap
createActionCreator('B', resolve => (error: Error) => resolve(error))

// @dts-jest:pass:snap
createActionCreator('C', resolve => (error: Error, meta?: { status: number }) =>
  resolve(error, meta)
)

// @dts-jest:pass:snap
createActionCreator(
  'D',
  // tslint:disable-next-line:no-inferrable-types
  resolve => (name: string, completed: boolean = false) =>
    resolve({ name, completed })
)

// @dts-jest:pass:snap
createActionCreator(
  'E',
  // tslint:disable-next-line:no-inferrable-types
  resolve => (name: string, completed: boolean = false) =>
    resolve({ name, completed }, 'Meta')
)

// @dts-jest:pass:snap
createActionCreator('F', resolve => <TName>(name: TName) => resolve(name))

// @dts-jest:pass:snap
createActionCreator('G', resolve => <TName, TMeta>(name: TName, meta: TMeta) =>
  resolve(name, meta)
)

// @dts-jest:pass:snap
createActionCreator('H', resolve => (...names: string[]) => resolve(names))
