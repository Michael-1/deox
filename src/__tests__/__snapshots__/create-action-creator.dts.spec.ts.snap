// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createActionCreator createActionCreator(
      'D',
      // tslint:disable-next-line:no-inferrable-types
      resolve => (name: string, completed: boolean = false) =>
        resolve({ name, completed })
    ) (type) should match snapshot 1`] = `"ExactActionCreator<\\"D\\", (name: string, completed?: boolean) => { type: \\"D\\"; payload: { name: string; completed: boolean; }; }>"`;

exports[`createActionCreator createActionCreator(
      'E',
      // tslint:disable-next-line:no-inferrable-types
      resolve => (name: string, completed: boolean = false) =>
        resolve({ name, completed }, 'Meta')
    ) (type) should match snapshot 1`] = `"ExactActionCreator<\\"E\\", (name: string, completed?: boolean) => { type: \\"E\\"; payload: { name: string; completed: boolean; }; meta: string; }>"`;

exports[`createActionCreator createActionCreator('A') (type) should match snapshot 1`] = `"ExactActionCreator<\\"A\\", () => { type: \\"A\\"; }>"`;

exports[`createActionCreator createActionCreator('B', resolve => (error: Error) => resolve(error)) (type) should match snapshot 1`] = `"ExactActionCreator<\\"B\\", (error: Error) => { type: \\"B\\"; payload: Error; error: true; }>"`;

exports[`createActionCreator createActionCreator('C', resolve => (error: Error, meta?: { status: number }) =>
      resolve(error, meta)
    ) (type) should match snapshot 1`] = `"ExactActionCreator<\\"C\\", (error: Error, meta?: { status: number; } | undefined) => { type: \\"C\\"; payload: Error; meta: { status: number; }; error: true; }>"`;

exports[`createActionCreator createActionCreator('F', resolve => <TName>(name: TName) => resolve(name)) (type) should match snapshot 1`] = `"ExactActionCreator<\\"F\\", <TName>(name: TName) => Action<\\"F\\", TName, undefined>>"`;

exports[`createActionCreator createActionCreator('G', resolve => <TName, TMeta>(name: TName, meta: TMeta) =>
      resolve(name, meta)
    ) (type) should match snapshot 1`] = `"ExactActionCreator<\\"G\\", <TName, TMeta>(name: TName, meta: TMeta) => Action<\\"G\\", TName, TMeta>>"`;

exports[`createActionCreator createActionCreator('H', resolve => (...names: string[]) => resolve(names)) (type) should match snapshot 1`] = `"ExactActionCreator<\\"H\\", (...names: string[]) => { type: \\"H\\"; payload: string[]; }>"`;
